File: multiplesof3or5/python/blackprince.py
# author @blackprince001

def natural_mod_3_5(n: int) -> list[int]:
    nums = []
    for i in range(1, n+1):
        if i % 3 == 0 or i % 5 == 0:
            nums.append(i)
    return nums

def sum_elements(arr: list[int]) -> int:
    return sum(arr)


if __name__ == "__main__":
    to_1000 = natural_mod_3_5(1000)
    sum_to_1000 = sum_elements(to_1000)

    print(f"Sum of multiples of 3 and 5 from 1 to 1000 is {sum_to_1000}")

File: multiplesof3or5/cpp/blackprince.cpp
// Author: @blackprince001

#include <iostream>
#include <vector>

std::vector<int> multiples_of_3and5(int nth) {
    std::vector<int> mul_collection;
    for (int i = 11; i <= nth; ++i)
        // check if number is divisible by 3 or 5 using ternary operations
        mul_collection.push_back((i % 3 == 0 || i % 5 == 0) ? i : 0);
    return mul_collection;
}

int main() {
    // initialize the `sum` and `limit` variables
    int sum = 0;
    int limit = 1000;

    // generate the list of multiples of 3 and 5 and store in `list`
    auto list = multiples_of_3and5(limit);

    // iterate over `list` and add those values to the cumulative `sum`
    for (auto &val : list) sum += val;

    // print the sum
    std::cout << "Sum of values is " << sum;

    return 0;
}